build/OutputDirectories.props

```diff
@@ -1,23 +1,35 @@
 <Project>
   <PropertyGroup>
-      <BaseOutputDirectory>$(RepoRoot)/artifacts/$(Rid)</BaseOutputDirectory>
-      <OutputDirectory>$(BaseOutputDirectory)/stage2</OutputDirectory>
-      <Stage2WithBackwardsCompatibleRuntimesOutputDirectory>$(BaseOutputDirectory)/stage2WithBackwardsCompatibleRuntimes</Stage2WithBackwardsCompatibleRuntimesOutputDirectory>
+      <CliBuildStage Condition="'$(CliBuildStage)' == '' And '$(CliOuterBuildStage)' != ''">$([MSBuild]::Add($(CliOuterBuildStage), '1'))</CliBuildStage>
+      <CliBuildStage Condition="'$(CliBuildStage)' == ''">2</CliBuildStage>
+
+      <!-- BaseOutputDirectory is the root output path for this stage and RID
+           BaseOutputPath is the project-specific output folder that each project builds into.
+           This is not confusing at all. -->
+      <BaseOutputDirectory>$(RepoRoot)/out/$(CliBuildStage)/$(Rid)</BaseOutputDirectory>
```

Comment:
Can the `out` directory be renamed to `bin` to align with the rest of the dotnet repos?

Reply:
@eerhardt There are also `bin` directories further down under `out` (ie `out\2\win10-x64\bin\<ProjectName>`), and the intermediate (`obj`) folders are traditionally a peer to `bin`, but they are also under the `out` folder.  Given that, would you still prefer `bin` for the top-level output directory?

Reply:
Yes, it is consistent with the rest of the repos.  `corefx`, `coreclr`, `core-setup`, `standard`, `sdk`, `BuildTools` all use `bin` at the root of the repo.

These repos all make the intermediate output folder under `bin/obj`.

The term `artifacts` mostly came from ASP.NET and aligns with their repos.

If we are going to change this folder's name, we should change it to align with the majority of the `dotnet` repos.  In general, we need to be more consistent across our repos to make it easier for other team members to contribute across repos.